### PR TITLE
Javascript javascript1 week2

### PR DESCRIPTION
**Pull request by MoeinHeydari90**
These items are included in the code:

1. Flight booking full name function
2. Event application
3. Weather wear
4. Student manager
5. Candy helper

### CODE REVIEW

File: javascript/javascript1/week2/script.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// Flight booking fullname function
+
+function getFullname(firstname, surname, sex, useFormalName = false) {
+    let prefix = "";
+
+    if (useFormalName) {
+        if (sex === "male") prefix = "Lord";
+        else if (sex === "female") prefix = "Lady";
+        else console.log("Please enter a valid gender");
+    }
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/script.js:12]: Really nice logic & very easy to read!

File: javascript/javascript1/week2/script.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// * My free code camp link : https://www.freecodecamp.org/fcc733a70bc-2aaa-4346-b4d1-7b85038190f9
+
+// Flight booking fullname function
+
+function getFullname(firstname, surname, sex, useFormalName = false) {
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/script.js]: Only fix I can see is remember to use camelCase (even though I know for this assignment it is spelt incorrectly)

File: javascript/javascript1/week2/script.js [hunk 3]
```diff
@@ -115,3 +115,52 @@ console.log(
...
+const candies = Object.keys(candyTablePrice); // To get the list of candies from the candies price table
+
+while (canBuyMoreCandy()) {
+    // To choose a random candy each time
+    let randomIndex = Math.floor(Math.random() * candies.length);
+    let randomCandy = candies[randomIndex];
+
+    // Adding candy until the budget is over
+    addCandy(randomCandy, Math.round(Math.random() * 50 + 1));
+}
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/script.js:166]: Really nice detail adding this logic!

File: javascript/javascript1/week2/script.js [hunk 4]
```diff
@@ -115,3 +115,52 @@ console.log(
...
+// Candy helper
+
+let boughtCandyPrices = [];
+
+const candyTablePrice = {
+    sweet: 0.5,
+    chocolate: 0.7,
+    toffee: 1.1,
+    "chewing-gum": 0.03,
+};
```
Comment (islanderupnorth) [inline javascript/javascript1/week2/script.js:128]: Great you've used this approach of storing the values in an object

**Review by islanderupnorth — COMMENTED** [review]
Really good work Moein! You seem to have a very logical approach and write good clean code. I must also compliment the comments you have sprinkled throughout your code. Keep that good habit up!

**Comment by MoeinHeydari90** [comment]
Thank you for reviewing my code Nicolai

**Pull request merged**